From c74ee2d88ff25cdfd88f97c4ec56a481d43c0c57 Mon Sep 17 00:00:00 2001
From: Yehuda Yitschak <yehuday@marvell.com>
Date: Wed, 17 Jun 2015 14:29:28 +0300
Subject: [PATCH 0200/1240] Kconfig: setup infrastructure for devel board
 support

These added defconfigs will be used to compile out
development board support

Change-Id: Idc9dc5ca132c6bac72fa5a2e32af59876ade250f
Signed-off-by: Yehuda Yitschak <yehuday@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/20248
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm/Kconfig                  |  1 +
 board/mvebu/common/Kconfig        | 32 ++++++++++++++++++++++++++++++++
 board/mvebu/common/Makefile       |  2 +-
 board/mvebu/common/init.c         |  4 ++--
 configs/mvebu_armada38x_defconfig |  2 ++
 configs/mvebu_armada8k_defconfig  |  4 +++-
 configs/mvebu_armadalp_defconfig  |  1 +
 include/configs/armada38x.h       |  3 ---
 include/configs/armadalp.h        |  3 ---
 include/configs/mvebu-common.h    |  2 +-
 10 files changed, 43 insertions(+), 11 deletions(-)
 create mode 100644 board/mvebu/common/Kconfig

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index f6b760b..422c46d 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -852,6 +852,7 @@ source "arch/arm/cpu/armv7/zynq/Kconfig"
 source "arch/arm/cpu/armv7/Kconfig"
 
 source "arch/arm/cpu/mvebu-common/Kconfig"
+source "board/mvebu/common/Kconfig"
 
 source "board/aristainetos/Kconfig"
 source "board/BuR/kwb/Kconfig"
diff --git a/board/mvebu/common/Kconfig b/board/mvebu/common/Kconfig
new file mode 100644
index 0000000..c7b69d5
--- /dev/null
+++ b/board/mvebu/common/Kconfig
@@ -0,0 +1,32 @@
+
+menuconfig DEVEL_BOARD
+	bool "Support MVEBU development boards"
+	depends on MVEBU
+	default n
+	---help---
+	  Say Y here to add support for Marvell MVEBU
+	  development boards.
+	  You need this only of you work on a Marvell board.
+	  If not, keep this off to reduce code size
+
+if DEVEL_BOARD
+
+	config ARMADA_38X_DB
+	bool "Support Marvell Armada-38x DB board"
+	default n
+	help
+	  Choose this option to add
+	  support for Marvell Armada-38x
+	  DB board. Say N unless you really
+	  work on this board
+
+	config APN_806_DB
+	bool "Support Marvell APN-806 DB board"
+	default n
+	help
+	  Choose this option to add
+	  support for Marvell APN-806
+	  DB board. Say N unless you really
+	  work on this board
+
+endif #DEVEL_BOARD
diff --git a/board/mvebu/common/Makefile b/board/mvebu/common/Makefile
index 94ca66e..4899d48 100644
--- a/board/mvebu/common/Makefile
+++ b/board/mvebu/common/Makefile
@@ -24,7 +24,7 @@ obj-$(CONFIG_MVEBU_NAND) += nand_init.o
 obj-$(CONFIG_MVEBU_PCI)	+= pcie_init.o
 obj-$(CONFIG_CMD_NET)	+= eth_init.o
 
-ifdef CONFIG_MVEBU_DEVEL_BOARD
+ifdef CONFIG_DEVEL_BOARD
 obj-y	+= sar.o
 obj-y	+= config.o
 obj-y	+= devel-board.o
diff --git a/board/mvebu/common/init.c b/board/mvebu/common/init.c
index 3faa169..63fb52b 100644
--- a/board/mvebu/common/init.c
+++ b/board/mvebu/common/init.c
@@ -29,7 +29,7 @@
 #include <asm/arch-mvebu/pinctl.h>
 #include <asm/arch-mvebu/mpp.h>
 #include "board.h"
-#ifdef CONFIG_MVEBU_DEVEL_BOARD
+#ifdef CONFIG_DEVEL_BOARD
 #include "devel-board.h"
 #endif
 
@@ -148,7 +148,7 @@ int mvebu_board_init(void)
 	mpp_bus_probe();
 #endif
 
-#ifdef CONFIG_MVEBU_DEVEL_BOARD
+#ifdef CONFIG_DEVEL_BOARD
 	mvebu_devel_board_init(brd_fam);
 #endif
 
diff --git a/configs/mvebu_armada38x_defconfig b/configs/mvebu_armada38x_defconfig
index 2dd3755..ff28c54 100644
--- a/configs/mvebu_armada38x_defconfig
+++ b/configs/mvebu_armada38x_defconfig
@@ -1,6 +1,8 @@
 CONFIG_SPL=y
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_38X=y
++S:CONFIG_DEVEL_BOARD=y
++S:CONFIG_ARMADA_38X_DB=y
 CONFIG_MVEBU_NAND=y
 CONFIG_MVEBU_SPI=y
 CONFIG_MVEBU_PCI=y
diff --git a/configs/mvebu_armada8k_defconfig b/configs/mvebu_armada8k_defconfig
index 8acd44b..5da26e3 100644
--- a/configs/mvebu_armada8k_defconfig
+++ b/configs/mvebu_armada8k_defconfig
@@ -2,8 +2,9 @@ CONFIG_SPL=y
 CONFIG_SYS_EXTRA_OPTIONS="ARM64"
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_8K=y
++S:CONFIG_DEVEL_BOARD=y
+CONFIG_APN_806_DB=y
 CONFIG_MVEBU_MPP=y
-+S:CONFIG_SYS_NS16550=y
 +S:CONFIG_MVEBU_MK6_SUPPORT=y
 CONFIG_CMD_BOOTD=y
 CONFIG_CMD_RUN=y
@@ -16,3 +17,4 @@ CONFIG_CMD_TIMER=y
 +S:CONFIG_OF_CONTROL=y
 +S:CONFIG_OF_EMBED=y
 +S:CONFIG_DEFAULT_DEVICE_TREE="armada-8k"
++S:CONFIG_SYS_NS16550=y
diff --git a/configs/mvebu_armadalp_defconfig b/configs/mvebu_armadalp_defconfig
index 692395f..3a735f6 100644
--- a/configs/mvebu_armadalp_defconfig
+++ b/configs/mvebu_armadalp_defconfig
@@ -1,6 +1,7 @@
 CONFIG_SPL=y
 +S:CONFIG_ARM=y
 +S:CONFIG_TARGET_ARMADA_LP=y
++S:CONFIG_DEVEL_BOARD=y
 +S:CONFIG_MVEBU_UART_ARLP=y
 CONFIG_CMD_BDI=y
 CONFIG_CMD_BOOTD=y
diff --git a/include/configs/armada38x.h b/include/configs/armada38x.h
index ee598a2..5a44cdb 100644
--- a/include/configs/armada38x.h
+++ b/include/configs/armada38x.h
@@ -39,9 +39,6 @@
 /* Version */
 #define CONFIG_IDENT_STRING	" Marvell version: 2014_T2.0"
 
-/* Configurable boards */
-#define CONFIG_MVEBU_DEVEL_BOARD      /* Turn this off for Customer boards */
-
 /* Platform */
 #define CONFIG_MVEBU_CA9
 #define CONFIG_MVEBU_TIMER
diff --git a/include/configs/armadalp.h b/include/configs/armadalp.h
index 145fa86..0034a12 100644
--- a/include/configs/armadalp.h
+++ b/include/configs/armadalp.h
@@ -23,9 +23,6 @@
 #define CONFIG_IDENT_STRING		" Marvell version: 2015_T2.0"
 #define CONFIG_BOOTP_VCI_STRING		" U-boot.armv7.armadalp"
 
-/* Configurable boards */
-#define CONFIG_MVEBU_DEVEL_BOARD      /* Turn this off for Customer boards */
-
 /* Platform */
 #define CONFIG_MVEBU_CA9
 #define CONFIG_MVEBU_TIMER
diff --git a/include/configs/mvebu-common.h b/include/configs/mvebu-common.h
index 4f6d129..7d4e542 100644
--- a/include/configs/mvebu-common.h
+++ b/include/configs/mvebu-common.h
@@ -262,7 +262,7 @@
 	/*#define CONFIG_MVEBU_PCI_EP*/ /* Set PCI host as end point */
 
 	/* Enable PCIE NIC for devel boards */
-	#ifdef CONFIG_MVEBU_DEVEL_BOARD
+	#ifdef CONFIG_DEVEL_BOARD
 		#define CONFIG_EEPRO100
 		#define CONFIG_E1000
 		#define CONFIG_MVEBU_NET
-- 
1.9.1

